DOC Add more examples to Series.Resample docstring

Winterflower opened the pull request:
Partial fix for https://github.com/pydata/pandas/issues/5023
flake8 doesn't do a very good job at checking code in the docstrings for PEP8 compliance, so I'm just eyeballing it at the moment and my PEP8 knowledge leaves something to be desired. Apologies if I missed something really obvious.

sinhrks commented on pandas/core/generic.py:
```diff
@@ -3274,7 +3274,50 @@ def resample(self, rule, how=None, axis=0, fill_method=None,
+
+        >>> index = pd.date_range('1/1/2000', periods=9, freq='T')
+        >>> series = pd.Series(range(9), index=index)
+        >>> print series
```
Remove ``print`` (for py3), only variable name (``series``) is better.

sinhrks commented on pandas/core/generic.py:
```diff
@@ -3274,7 +3274,50 @@ def resample(self, rule, how=None, axis=0, fill_method=None,
+        Downsample the series into 3 minute bins as above, but label each
+        bin using the right edge instead of the left. 
+
+        >>> series.resample('3T', how='sum', label='left')
```
``label='right'``?

Winterflower replied:
@sinhrks ah yes, sloppy me! I must have correct it in the Ipython notebook I use to check these and forgotten to place it in the file. Thanks!

Winterflower commented:
Will squash everything into one commit once I add an example of using a custom ``how`` method.

Winterflower committed:
Add more examples to Series.resample

Add more resample examples

Improve Series examples

Series resample fixes

Fixed breaking docstring issue

Winterflower commented:
@jorisvandenbossche @sinhrks 
this one is done and good to go from my side!
comments/corrections welcome, of course!

jorisvandenbossche commented on pandas/core/generic.py:
```diff
@@ -3274,7 +3274,90 @@ def resample(self, rule, how=None, axis=0, fill_method=None,
+        Downsample the series into 3 minute bins as above, but label each
+        bin using the right edge instead of the left. 
+
+        >>> series.resample('3T', how='sum', label='right')
```
Maybe you also add `how=right'` here as well? (I always find that in most  common cases you need to specify both)

Winterflower replied:
@jorisvandenbossche I'm trying to test some examples with ``how=right`` and I am getting exceptions. What is '``how=right`` supposed to do?

jorisvandenbossche replied:
oops, sorry, I meant 'closed' instead of 'how'

Winterflower replied:
haha no worries! ``how=right`` did seem a bit weird. I'll include ``closed=right`` in the example.

jorisvandenbossche commented:
Nice examples!

sinhrks commented:
Looks nice! No further comments from me.

jorisvandenbossche commented:
@Winterflower Do you have some time to update? (it's only a very small comment)

Winterflower commented:
@jorisvandenbossche apologies for the delay! I will add the how='right' example tonight after work.

Winterflower commented:
Still need to fix line lengths.

Winterflower commented:
At the moment, the doctest for the string is failing because of the ``custom_resampler`` function defined in the docstring

```
Trying:
    def custom_resampler(array_like):
Expecting:
            return np.sum(array_like)+5
**********************************************************************
File "pandas/core/generic.py", line 3367, in generic.NDFrame.resample
Failed example:
    def custom_resampler(array_like):
Exception raised:
    Traceback (most recent call last):
      File "/home/winterflower/anaconda/lib/python2.7/doctest.py", line 1315, in __run
        compileflags, 1) in test.globs
      File "<doctest generic.NDFrame.resample[9]>", line 1
        def custom_resampler(array_like):
                                        ^
    SyntaxError: unexpected EOF while parsing
```

Winterflower commented:
This is good to go on my part!

jreback commented:
@Winterflower gr8 thanks!

Further I think portions of your narrative my be a nice addition to the actual resample docs; add where appropriate
